Runtime/BacktraceClient.cs

```diff
@@ -991,7 +991,19 @@ internal void HandleUnhandledExceptionsFromAndroidBackgroundThread(string backgr
             }
             var message = backgroundExceptionMessage.Substring(0, splitIndex);
             var stackTrace = backgroundExceptionMessage.Substring(splitIndex);
-            HandleUnityMessage(message, stackTrace, LogType.Exception);
+            if (Database != null)
+            {
+                Database.Add(new BacktraceReport(new BacktraceUnhandledException(message, stackTrace)).ToBacktraceData(null, GameObjectDepth));
+            }
+            else
+            {
+                HandleUnityMessage(message, stackTrace, LogType.Exception);
+            }
+            var androidNativeClient = _nativeClient as Runtime.Native.Android.NativeClient;
```

Comment:
Eww. Casting. Do we need to? :) Can we fix in a different way on mainline?

Reply:
As in ok for bugfix, but probably want to improve leaky abstraction here

Reply:
this is a special method that is available only for android - other native clients don't have it. This casting is safe, but I want to be sure it's always safe.